### DISCUSSION TITLE
How to handle R that's not symmetric

### DISCUSSION THREAD

**Discussion by ansetou** [qpsolvers/qpsolvers#92] [Q&A] [answered]
Hi, I'm trying to use qpsolver with quad_prog library to solve a linear least square problem. I'm very new to this and mostly just following online instructions. My question is if I have m data point equations and n variables. and m != n. then my R will not be symmetric. How do I apply W that's not None?

**Comment by stephane-caron** [comment, marked as answer]
There was a mistake in the doc, sorry for the confusion.

If is fine if $R$ is not symmetric. The main requirement, for most solvers, will be that $R$ has full row rank.

I've updated the doc:

* Previously the doc described $R$ as: "Symmetric matrix of the cost function (most solvers require it to be definite)."
* That was a copy-paste error. Now the doc describes $R$ as: "Matrix factor of the cost function (most solvers require $R^T W R$ to be positive definite, which means $R$ should have full row rank)."
